src/mpl/src/gavl/mpl_gavl.c

```diff
@@ -14,47 +14,47 @@
  * h_p - 2 height right child; therefore, the worse case nodes N(h_p) = N(h_p - 1) + N(h_p - 2) + 1.
  * Since we know N(1) = 1 and N(2) = 2, we can use iteration to compute N(64) = 27777890035287.
  */
-#define MAX_STACK_SIZE 64
+#define MPLI_GAVL_MAX_STACK_SIZE 64
 
 enum {
-    SEARCH_LEFT,
-    SEARCH_RIGHT,
-    BUFFER_MATCH,
-    NO_BUFFER_MATCH
+    MPLI_GAVL_SEARCH_LEFT,
+    MPLI_GAVL_SEARCH_RIGHT,
+    MPLI_GAVL_BUFFER_MATCH,
+    MPLI_GAVL_NO_BUFFER_MATCH
 };
 
 enum {
-    SUBSET_SEARCH,
-    INTERSECTION_SEARCH
+    MPLI_GAVL_SUBSET_SEARCH,
+    MPLI_GAVL_INTERSECTION_SEARCH
 };
 
-typedef struct gavl_tree_node {
+typedef struct MPLI_gavl_tree_node {
     union {
         struct {
-            struct gavl_tree_node *parent;
-            struct gavl_tree_node *left;
-            struct gavl_tree_node *right;
+            struct MPLI_gavl_tree_node *parent;
+            struct MPLI_gavl_tree_node *left;
+            struct MPLI_gavl_tree_node *right;
         };
```

Comment:
I know this is separate, but would you mind fix the anonymous union/struct usage? It is a C11 feature and we only require C99. Sun compiler in particular throws warning for it right now. It is not important, so it's ok if it is too out-of-your way.

Reply:
I tried to fix it by adding name for the anonymous union/struct. But the code changes are a bit messy as we have to change all pointer accesses in GAVL. It might be better to separate the fix into a different PR